sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestClassShouldHaveTestMethod.cs

```diff
@@ -61,9 +61,11 @@ protected override void Initialize(SonarAnalysisContext context)
                     }
 
                     var classDeclaration = (ClassDeclarationSyntax)c.Node;
-                    var classSymbol = c.SemanticModel.GetDeclaredSymbol(classDeclaration);
+                    var classSymbol = c.SemanticModel.GetDeclaredSymbol(classDeclaration);
+
                     if (classSymbol == null ||
-                        !classSymbol.GetAttributes().Any(a => a.AttributeClass.IsAny(HandledTestClassAttributes)))
+                        !classSymbol.GetAttributes().Any(a => a.AttributeClass.IsAny(HandledTestClassAttributes)) ||
```

Comment:
Move IsAbstract check before the attribute check. Hey, am I suggesting micro-performace tweak? :)